### PR TITLE
230 basic linting rule sim

### PR DESCRIPTION
**Pull request by mschwoer**
-   add rule "SIM" to ruleset
  -  make code pass rule "SIM" (see individual commits)

### CODE REVIEW

File: alphadia/plexscoring.py [hunk 1]
```diff
@@ -1111,7 +1111,7 @@ def get_feature_columns(self):
                 if len(candidate.features) not in known_feature_lengths:
                     known_feature_lengths += [len(candidate.features)]
                     # add all new features to the list of known columns
-                    for key in candidate.features.keys():
+                    for key in candidate.features.keys():  # noqa: SIM118
```
Comment (GeorgWa) [inline alphadia/plexscoring.py:1114]: why does this require a noqa?

File: alphadia/workflow/peptidecentric.py [hunk 2]
```diff
@@ -503,17 +499,16 @@ def calibration(self):
...
-                self.reporter.log_string(
-                    "Performing final calibration with all precursors",
-                    verbosity="progress",
-                )
-                features_df, fragments_df = self.extract_batch(
-                    self.spectral_library._precursor_df
-                )
-                precursor_df = self.fdr_correction(features_df, fragments_df)
-                self.recalibration(precursor_df, fragments_df)
+        if self.config["calibration"].get("final_full_calibration", False):
```
Comment (GeorgWa) [inline alphadia/workflow/peptidecentric.py:502]: cooles pattern!

File: release/pyinstaller/cli_hook.py [hunk 3]
```diff
@@ -8,12 +8,11 @@
     # catch pyinstaller bug
     # subprocess.CalledProcessError: Command '['/Applications/alphaDIA.app/Contents/Frameworks/alphadia', '-sS', '-c', 'import platform; print(platform.mac_ver()[0])']' returned non-zero exit status 2.
     # [36481] Failed to execute script 'cli_hook' due to unhandled exception!
-    if len(sys.argv) >= 3:
-        if (sys.argv[1] == "-sS") & (sys.argv[2] == "-c"):
-            import platform
+    if len(sys.argv) >= 3 and (sys.argv[1] == "-sS") & (sys.argv[2] == "-c"):
```
Comment (GeorgWa) [inline release/pyinstaller/cli_hook.py:11]: what is the binary operator order with python?

**Review by GeorgWa — APPROVED** [review]
LGTM

File: release/pyinstaller/cli_hook.py (hunk 3 above)
Comment (mschwoer) [reply release/pyinstaller/cli_hook.py]: simplified it by replacing "&" with "and"

**Pull request merged**